django/thunderstore/modpacks/tasks.py

```diff
@@ -0,0 +1,50 @@
+import base64
+import io
+import zipfile
+import yaml
+
+from celery import shared_task
+from typing import Dict
+from thunderstore.core.settings import CeleryQueues
+from thunderstore.modpacks.models.legacyprofile import LegacyProfile, LegacyProfileMetaData
+
+def parse_package_name_and_team(s: str):
+    reversed_splitted = "".join(reversed(s)).split("-", maxsplit=1)
+    name = "".join(reversed(reversed_splitted[0]))
+    team = "".join(reversed(reversed_splitted[1]))
+    return (name, team)
+
+
+def generate_mod_metadata(mod: Dict):
+    name, team = parse_package_name_and_team(mod["name"])
+    version = ("%s.%s.%s" % (mod["version"]["major"], mod["version"]["minor"], mod["version"]["patch"]))
+    return {
+        "name": name,
+        "team": team,
+        "version": version,
+        "enabled": mod["enabled"],
+        "url": ("https://thunderstore.io/c/valheim/p/%s/%s/v/%s" % (
```

Comment:
Might be best to not store an URL at all given that the URL will eventually change, and this is intended to be a static cache. The consumer of this data should know how to build the URL given the appropriate namespace, team, and version.

Now that I mention namespace, rename the `team` field to `namespace` while you're at it.